lib/cloudregister/registerutils.py

```diff
@@ -516,6 +517,56 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    """Set all the necessary parts for the registry,
+       returns True if the setup completed, False otherwise."""
+    if not os.path.exists(os.path.dirname(REGISTRY_CREDENTIALS_PATH)):
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH))
+
+    setup_registry_succeed = set_registry_auth_token(
+        registry_fqdn, username, password
+    )
+    return setup_registry_succeed
+
+
+# ----------------------------------------------------------------------------
+def set_registry_auth_token(registry_fqdn, username, password):
+    """Set the auth token to pull images from SUSE registry."""
+    auth_token = base64.b64encode('{username}:{password}'.format(
+        username=username,
+        password=password
+    ).encode()).decode()
+    registry_credentials = {registry_fqdn: {'auth': auth_token}}
+    config_json = {}
+    try:
```

Comment:
wouldn't that be a one time occurrence ? 
and after any successful subsequent de-registration + registration the file would be present, right ?

Reply:
also, in the scenario of subsequent registrations, wouldn't it be better to not check with an if, in case the file gets removed between the check and the actual reading ?

Reply:
Maybe it is not an exception per se, but depending on the use, the file being present is more likely to be the common case
Still think it is better to deal with files in a better way than an if but  the likelihood of the file being remove may be low, guess we can change it if that scenario ever showed up